Automattic/wp-calypso
# Add @automattic/onboarding package to standardize stylesheets & reuse components.

## Changes proposed in this Pull Request

1. **Created `@automattic/onboarding` package.**
  The intention of this package is to offer onboarding stylesheets & components that can be reused across Calypso, FSE & WP-Admin. Currently it has SCSS variables & mixins, `<Title>` and `<Subtitle>` component taken out from Gutenboarding.

2. **Expanded `@automattic/typography` with font family & font size variables.**
  This is taken out from `client/assets/stylesheets/shared/_typography.scss` file so it can be used outside of Calypso.

3. **Clean up all the SCSS variables & mixins, `<Title>` & `<Subtitle>` duplication across the calypso codebase (within the scope of onboarding & typography). This includes:**
   - Updating `gutenboarding` to use `@automattic/onboarding`.
   - Updating `@automattic/domain-picker` to use `@automattic/onboarding`.
   - Updating `@automattic/plans-grid` to use `@automattic/onboarding`.
   - Updating `magic-login`, `wp-login` & `signup` stylesheet to use mixins from `@automattic/onboarding`.
   - Updating `wpcom-block-editor-nux` plugin to import fonts & mixins from `@automattic/typography` & 
`@automattic/onboarding`.

There are a lot of files modified in this PR. However, they do very simple changes. To review this PR, it is easier to review by commits, as commits are grouped thematically. Most work revolves around importing, renaming variables and removing duplicated codes.

## Usage Examples

For using onboarding/typography variables & mixins:
```scss
@import '~@automattic/onboarding/styles/variables';
@import '~@automattic/onboarding/styles/mixins';
@import '~@automattic/typography/styles/fonts';
@import '~@automattic/typography/styles/variables';
```

For using `<Title>` & `<SubTitle>` components:
```js
@import { Title, SubTitle } from '@automattic/onboarding';
```

## Testing instructions

* Run `yarn` to rebuild all packages.
* Run `yarn start` to boot up Calypso locally.
* As long as there is no compilation errors, all variables & mixins have been imported successfully.
* Test gutenboarding on `/new`, everything should look the same.
* Look around Calypso, everything should look the same.

Note: Testing launch flow on FSE is on this PR https://github.com/Automattic/wp-calypso/pull/44087 which is based off this PR.

Fixes part of #43750
